src/views/chart.js

```diff
@@ -63,31 +63,51 @@ const testData = [
 ];
 
 function ChartView({ ceoName, companyName }) {
-  const [collaborators, setCollaborators] = React.useState([
-    {
-      id: 0,
-      name: ceoName,
-      position: "CEO",
-      parent: "",
-      children: [1, 2, 3]
-    },
-    ...testData
-  ]);
+  const initialCollaborators = JSON.parse(
+    localStorage.getItem("collaborators") ||
+      JSON.stringify([
+        {
+          id: 0,
+          name: ceoName,
+          position: "CEO",
+          parent: "",
+          children: [1, 2, 3]
+        },
+        ...testData
+      ])
+  );
+
+  const [collaborators, setCollaborators] = React.useState(
+    initialCollaborators
+  );
 
   React.useEffect(() => {
     localStorage.setItem("collaborators", JSON.stringify(collaborators));
   }, [collaborators]);
 
+  function addCollaborator(collaboratorData) {
+    setCollaborators(state => {
+      state[collaboratorData.parent].children.push(state.length);
```

Comment:
hi, can you please explain what this function does? because I'm currently trying out your app and it crashes when I try adding a new child to another child (third line, "grandson" of the CEO), and the source of the problem is this line/function